packages/utils/other-utils/src/set-path-utils.js

```diff
@@ -0,0 +1,139 @@
+/**
+ * Depth-first search of cell set tree, starting from node.
+ * @param {*} node The node to start from.
+ * @param {*} currentPath The current path.
+ * @param {*} paths Accumulated paths so far.
+ * @returns all paths from node to leaves.
+ */
+function getPaths(node, currentPath = [], paths = []) {
+  if (node.children) {
+    node.children.forEach((child) => {
+      const newPath = [...currentPath, child.name];
+      paths.push(newPath);
+      getPaths(child, newPath, paths);
+    });
+  }
+  return paths;
+}
+
+/**
+ * Finds either the longest subset or the longest superset of path in
+ * arrOfPaths.
+ * @param {array} arrOfPaths An array of paths.
+ * @param {array} path An array of strings, representing a path.
+ * @param {boolean} isSubset A boolean flag that indicates whether we are
+ * looking for the longest subset (true) or the longest superset (false).
+ * @returns The longest subset or superset of path in arrOfPaths.
+*/
+const findLongest = (arrOfPaths, path, isSubset) => {
+  let longest = null;
+  let longestLength = 0;
+
+  arrOfPaths.forEach((subArray) => {
+    const matchCount = subArray.filter((v, i) => v === path[i]).length;
+    if (
+      matchCount === (isSubset ? subArray.length : path.length)
+          && subArray.length > longestLength
+    ) {
+      longest = subArray;
+      longestLength = subArray.length;
+    }
+  });
+  if (longestLength > 0) {
+    return longest;
+  } if (isSubset) {
+    return [];
+  }
+  return false;
+};
+
+/**
+ * Generates paths by doing DFS of tree and filters out the ones that are irrelevant.
+ * Filtering is based on the hierarchy, currently expanded and currently selected cell sets.
+ * @param {object} tree The current tree.
+ * @param {object} hierarchy The hierarchy object.
+ * @param {array} cellSetExpansion An array of cell set paths that are expanded.
+ * @param {array} cellSetSelection An array of cell set paths that are selected.
+ * @returns An array of paths that should be displayed.
+ */
+export function generateCellSetPaths(
```

Comment:
I would suggest a more descriptive name - `generateCellSetPaths` at first glance evokes generation of new paths that did not exist previously but based on the comments something like `filterPathsByExpansionAndSelection` may fit better

Reply:
that's a much better name, thanks for the suggestion!